infra/elasticsearch/src/utils/index.ts

```
@@ -0,0 +1,188 @@
+    // https://www.elastic.co/guide/en/elasticsearch/reference/current/explicit-mapping.html
+    // we do not want to enfore a external naming convention here.
+    /* eslint-disable @typescript-eslint/naming-convention */
+    await client.indices.create({
+      body: {
+        mappings: mappings,
+        settings: {
+          index: {
+            analysis: {
+              analyzer,
+              char_filter,
+              filter,
+              tokenizer,
+            },
+          },
+          number_of_replicas: 1,
+          number_of_shards: 1,
+        },
+      },
+      index: indexName,
+    });
+    logger.info(`Index ${indexName} created.`);
+  } catch (error: unknown) {
+    logger.error("index create", error);
+    throw error;
+  }
+}
+
+export async function version({ client }: UtilsOptions): Promise<void> {
+  const { body } = await client.info<{ version: { number: string } }>();
+  logger.info(body.version.number);
+}
+
+export async function bulkIndexDocuments({
+  client,
+  indexName,
+  documents,
+}: UtilsOptions & { documents: Document[] }): Promise<void> {
+  try {
+    interface BulkApiResponse {
+      errors?: unknown;
+      items: { index: { status: number } }[];
+    }
+
+    const body = documents.flatMap((doc, i) => [
+      {
+        index: {
+          _index: indexName,
+          // if available, use our cdtnId as the actual Elastic document id
+          ...(doc.cdtnId && { _id: doc.cdtnId }),
+          // unless we're in testing mode where we use position
+          ...(process.env.NODE_ENV === "test" && { _id: i }),
+        },
+      },
+      doc,
+    ]);
+
+    const resp = await client.bulk<BulkApiResponse>({
+      body,
+      index: indexName,
+    });
+    if (resp.body.errors) {
+      const errorDocs = resp.body.items.filter(
+        (item) => item.index.status != 201
+      );
+      logger.error(`Errors during indexation : ${JSON.stringify(errorDocs)}`);
```

I think we can throw an error here.